cmd/snap-seccomp/main.go

```diff
@@ -516,23 +516,27 @@ func addSecondaryArches(secFilter *seccomp.ScmpFilter) error {
 		// 64bit code that would have to detect at runtime if
 		// it can be used or not. But we are told this is not
 		// rare with certain classes of embedded devices.
-		compatArch = arch.UbuntuKernelArchitecture()
+		switch arch.UbuntuKernelArchitecture() {
+		case "amd64":
+			compatArch = seccomp.ArchAMD64
+		case "arm64":
+			compatArch = seccomp.ArchARM64
+		case "ppc64el":
+			compatArch = seccomp.ArchPPC64LE
+		case "ppc64":
+			compatArch = seccomp.ArchPPC64
+		case "i386":
+			compatArch = seccomp.ArchX86
+		case "armhf":
+			compatArch = seccomp.ArchARM
+		case "powerpc":
+			compatArch = seccomp.ArchPPC
+
+		}
```

Comment:
Indeed, I added code for this now and also added tests for the compat architecture handling (unfortunately the tests are a bit tricky because of idiosyncrasies of libseccomp, see  https://github.com/seccomp/libseccomp/issues/86. This is also further complicated by the fact that libseccomp-golang has two architecture ints: one is a iota that is *only* used in libseccomp-golang. The other is the native C/kernel architecture (AUDIT_ARCH_*). There is no public API to map those. We now have 4 different arch mapping tables in the code :(